unittest/server/src/ivi_input_controller_uinttests.cpp

```diff
@@ -0,0 +1,1714 @@
+ *                         +# weston_pointer_start_grab() must be called once time
+ *                         +# weston_touch_start_grab() must be called once time
+ *                         +# wl_resource_post_event() must be called {MAX_NUMBER} times
+ */
+TEST_F(InputControllerTest, handle_seat_updated_caps_allAvailableWithNotSameAdrr)
+{
+    struct weston_keyboard *lpp_keyboard [] = {(struct weston_keyboard *)0xFFFFFFFF};
+    struct weston_pointer *lpp_pointer [] = {(struct weston_pointer *)0xFFFFFFFF};
+    struct weston_touch *lpp_touch [] = {(struct weston_touch *)0xFFFFFFFF};
+    SET_RETURN_SEQ(weston_seat_get_keyboard, lpp_keyboard, 1);
+    SET_RETURN_SEQ(weston_seat_get_pointer, lpp_pointer, 1);
+    SET_RETURN_SEQ(weston_seat_get_touch, lpp_touch, 1);
+
+    mpp_ctxSeat[CUSTOM_SEAT]->keyboard_grab.keyboard = nullptr;
+    mpp_ctxSeat[CUSTOM_SEAT]->pointer_grab.pointer = nullptr;
+    mpp_ctxSeat[CUSTOM_SEAT]->touch_grab.touch = nullptr;
+
+    handle_seat_updated_caps(&mpp_ctxSeat[CUSTOM_SEAT]->updated_caps_listener, &mp_westonSeat[CUSTOM_SEAT]);
+
+    ASSERT_EQ(1, weston_keyboard_start_grab_fake.call_count);
+    ASSERT_EQ(1, weston_pointer_start_grab_fake.call_count);
+    ASSERT_EQ(1, weston_touch_start_grab_fake.call_count);
+    ASSERT_EQ(MAX_NUMBER, wl_resource_post_event_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             handle_seat_updated_caps_notAvailable
+ * @brief               Test case of handle_seat_updated_caps() where weston_seat_get_keyboard() and weston_seat_get_pointer()
+ *                      and weston_seat_get_touch() are not mocked
+ * @test_procedure Steps:
+ *                      -# Calling the handle_seat_updated_caps() time 1
+ *                      -# Set input param keyboard_grab.keyboard to an object with address 0xFFFFFFFF
+ *                      -# Set input param pointer_grab.pointer to an object with address 0xFFFFFFFF
+ *                      -# Set input param touch_grab.touch to an object with address 0xFFFFFFFF
+ *                      -# Calling the handle_seat_updated_caps() time 2
+ *                      -# Verification point:
+ *                         +# weston_keyboard_start_grab() not be called
+ *                         +# weston_pointer_start_grab() not be called
+ *                         +# weston_touch_start_grab() not be called
+ *                         +# keyboard_grab.keyboard set to null pointer
+ *                         +# pointer_grab.pointer set to null pointer
+ *                         +# touch_grab.touch set to null pointer
+ *                         +# wl_resource_post_event() must be called {2*MAX_NUMBER} times
+ */
+TEST_F(InputControllerTest, handle_seat_updated_caps_notAvailable)
+{
+    handle_seat_updated_caps(&mpp_ctxSeat[CUSTOM_SEAT]->updated_caps_listener, &mp_westonSeat[CUSTOM_SEAT]);
```

Comment:
call two time of test function is not recommend. if can't split, need a detail comment